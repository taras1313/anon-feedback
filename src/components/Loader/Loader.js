import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const Loader = ({ height, width }) => {
	const useStyles = makeStyles(theme => ({
		root: {
			width: width || '100%',
			height: height || '100%',
			display: 'flex',
			justifyContent: 'center',
			alignItems: 'center',
			'& > * + *': {
				margin: 'auto'
			}
		}
	}));

	const classes = useStyles();

	return (
		<div className={classes.root}>
			<CircularProgress />
		</div>
	);
};

export default Loader;
