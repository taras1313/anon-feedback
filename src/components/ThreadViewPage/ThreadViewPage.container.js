import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { ThreadViewPage as ThreadView } from './ThreadViewPage';
import {
  setThreadData,
  subscribeToThread,
  unsubscribeFromThread,
  getThreadById,
  onCreateComment,
  onUpdateComment,
  like as likeThread,
  dislike as dislikeThread
} from '../../actions/threadActions';

const mapStateToProps = (
  {
    userReducer: { user: { _id } },
    threadsReducer: { selectedThread }
  }
) => ({ userId: _id, thread: selectedThread });

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators({
    setThreadData,
    subscribeToThread,
    unsubscribeFromThread,
    getThreadById,
    onCreateComment,
    onUpdateComment,
    likeThread,
    dislikeThread
  }, dispatch)
});

export const ThreadViewPage = connect(mapStateToProps, mapDispatchToProps)(ThreadView);

