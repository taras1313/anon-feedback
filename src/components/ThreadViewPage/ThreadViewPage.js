import React from 'react';

import { ThreadComponent } from '../TreadComponent';
import Loader from '../Loader/Loader';

export class ThreadViewPage extends React.Component {
	componentDidMount() {
		const {
			actions: { getThreadById } ,
			match: {
				params: { id }
			}
		} = this.props;
		getThreadById(id);
	}

	componentDidUpdate({match: {
		params: { id: prevId }
	}}) {

		const {
			actions: { getThreadById },
			match: {
				params: { id }
			}
		} = this.props;

		prevId !== id && getThreadById(id);
	}

	render() {
		const {
			thread,
			actions: { likeThread, dislikeThread, likeComment, dislikeComment }
		} = this.props;
		
		if (!thread) return <Loader />;

		return (
			<ThreadComponent likeThread={likeThread} dislikeThread={dislikeThread} likeComment={likeComment} dislikeComment={dislikeComment} threadView="full" {...this.props} />
		);
	}
}
