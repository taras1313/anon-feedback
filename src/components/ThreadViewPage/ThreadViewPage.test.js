import React from "react";
import { ThreadViewPage } from "./ThreadViewPage";
import { shallow } from "enzyme";
import Loader from '../Loader/Loader';

const props = {
  actions: { getThreadById: jest.fn(() => "newTestId") },
  match: {
    params: { id: "testId" },
  },
  thread: {},
};

describe("ThreadViewPage", () => {
  test("snapshot", () => {
    const wrapper = shallow(<ThreadViewPage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("did mount", () => {
    shallow(<ThreadViewPage {...props} />);

    expect(props.actions.getThreadById).toHaveBeenCalledWith("testId");
  });

  test("did mount", () => {
    const wrapper = shallow(<ThreadViewPage {...props} />);
    wrapper.setProps({
      match: {
        params: { id: "newTestId" },
      },
    });

    expect(props.actions.getThreadById).toHaveBeenCalledWith("newTestId");
  });

  test("no thread", () => {
    const wrapper = shallow(<ThreadViewPage {...props} thread={null} />);

    expect(wrapper.find(Loader)).toHaveLength(1);
  });
});
