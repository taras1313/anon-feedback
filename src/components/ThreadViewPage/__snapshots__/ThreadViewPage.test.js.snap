// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadViewPage snapshot 1`] = `
<ThreadComponent
  actions={
    Object {
      "getThreadById": [MockFunction] {
        "calls": Array [
          Array [
            "testId",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": "newTestId",
          },
        ],
      },
    }
  }
  answersCount={9}
  author={Object {}}
  dislikeComment={[Function]}
  dislikeThread={[Function]}
  dislikesCount={1}
  likeComment={[Function]}
  likeThread={[Function]}
  likesCount={9}
  match={
    Object {
      "params": Object {
        "id": "testId",
      },
    }
  }
  subscribersCount={3}
  thread={Object {}}
  threadView="full"
/>
`;
